### PR TITLE
Refactor Component initialization

### PR DESCRIPTION
**Pull request by caiofaustino**
## Summary
- Removed `PaymentComponentViewModelFactory` in favor of inline function with explicit constructor call.
- Removed `StoredPaymentMethodDelegate` in favor of explicit initialization.
- Moved some logic on the `CardComponent` to the new Delegate classes.
- Refactored the BCMC, GooglePay, WeChatPay Providers to Kotlin to use new factory.
- Refactored `GenericPaymentComponentProvider` to use old reflection initialization.

### CODE REVIEW

File: card/src/main/java/com/adyen/checkout/card/CardComponent.kt [hunk 1]
```diff
@@ -79,13 +78,19 @@ class CardComponent private constructor(
...
+        val firstCardType: CardType? = if (filteredSupportedCards.isNotEmpty()) filteredSupportedCards[0] else null
+
         return CardOutputData(
-            validateCardNumber(inputData.cardNumber),
-            validateExpiryDate(inputData.expiryDate),
-            validateSecurityCode(inputData.securityCode),
+            cardDelegate.validateCardNumber(inputData.cardNumber),
+            cardDelegate.validateExpiryDate(inputData.expiryDate),
+            cardDelegate.validateSecurityCode(inputData.securityCode, configuration, firstCardType),
             validateHolderName(inputData.holderName),
```
Comment (jreij) [inline card/src/main/java/com/adyen/checkout/card/CardComponent.kt:90]: Can this be `cardDelegate.validateHolderName` instead?

Comment (caiofaustino) [reply card/src/main/java/com/adyen/checkout/card/CardComponent.kt:90]: Yeah, this was not supposed to be part of this PR 😬 . Maybe I can finish this part and add it as well.

File: card/src/main/java/com/adyen/checkout/card/NewCardDelegate.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        cardType: CardType?
+    ): ValidatedField<String> {
+        return if (cardConfiguration.isHideCvc) {
+            ValidatedField(securityCode, ValidatedField.Validation.VALID)
+        } else {
+            CardValidationUtils.validateSecurityCode(securityCode, cardType)
+        }
+    }
+
+    override fun isCvcHidden(cardConfiguration: CardConfiguration): Boolean {
```
Comment (caiofaustino) [inline card/src/main/java/com/adyen/checkout/card/NewCardDelegate.kt:42]: @jreij do you think we should add the `CardConfiguration` to the constructor instead?

Comment (jreij) [reply card/src/main/java/com/adyen/checkout/card/NewCardDelegate.kt:42]: Hmm yeah why not

**Review by jreij — APPROVED** [review]
👍

**Pull request merged**